core/src/main/scala/kafka/server/metadata/MetadataPartitions.scala

```diff
@@ -198,15 +242,18 @@ class MetadataPartitionsBuilder(val brokerId: Int,
           }
           changed.add(newPartitionMap)
           newNameMap.put(topicName, newPartitionMap)
+          prevPartition
         }
       }
+      removedPartition.foreach(maybeAddToLocalRemoved)
     }
   }
 
   def build(): MetadataPartitions = {
-    val result = MetadataPartitions(newNameMap, newIdMap)
+    val result = new MetadataPartitions(newNameMap, newIdMap, newReverseIdMap)
```

Comment:
Is it worth checking consistency of those collections? It can produce quick failure if we are going to build invalid image.

Reply:
I would say yes if there was a cheap way to do it. However, I don't think it's unreasonable to depend on internally maintained invariants here.